serverlessworkflow/modules/ROOT/pages/security/orchestrating-third-party-services-with-oauth2.adoc

```diff
@@ -1 +1,700 @@
+<3> Specifies that the function returns a JSON object with a property `"calculateExchangeResult"` that contains the calculation.
+--
+
+To filter the information that must be returned to the workflow you can use a {spec_doc_action_data_filters_url}[CNCF Serverless Workflow actionDataFilter] such as:
+
+--
+.actionDataFilter to pass the `calculateExchange` results:
+[source, json]
+----
+  "actionDataFilter": {
+    "results": "${ {result: .calculateExchangeResult} }" <1>
+  }
+----
+<1> Merge the `"calculateExchangeResult"` property value coming from the expression result into a workflowdata property `"result"`.
+--
+
+== Acme Financial Service
+
+The following OpenAPI specification defines the `acme-financial-service`.
+
+--
+.acme-financial-service.yml OpenAPI specification
+[source#acme-financial-service-yml, yaml]
+----
+---
+openapi: 3.0.3
+info:
+  title: Acme Financial Service API
+  version: 1.0.1
+paths:
+  /financial-service/exchange-rate: <1>
+    get:
+      tags:
+        - Acme Financial Resource
+      operationId: exchangeRate <2>
+      parameters: <3>
+        - name: currencyFrom
+          in: query
+          schema:
+            type: string
+        - name: currencyTo
+          in: query
+          schema:
+            type: string
+        - name: exchangeDate
+          in: query
+          schema:
+            type: string
+      responses: <3>
+        "200":
+          description: OK
+          content: <4>
+            application/json:
+              schema:
+                $ref: '#/components/schemas/ExchangeRateResult'
+      security:
+        - acme-financial-oauth: [ ] <5>
+components:
+  schemas:
+    ExchangeRateResult: <6>
+      type: object
+      properties:
+        rate:
+          format: double
+          type: number
+  securitySchemes:
+    acme-financial-oauth: <7>
+      type: oauth2 <8>
+      flows:
+        clientCredentials: <9>
+          authorizationUrl: http://localhost:8281/auth/realms/kogito/protocol/openid-connect/auth
+          tokenUrl: http://localhost:8281/auth/realms/kogito/protocol/openid-connect/token
+          scopes: { }
+----
+
+<1> REST path to access the `exchangeRate` operation in the remote server.
+<2> `exchangeRate` operation parameters.
+<3> `exchangeRate` operation responses.
+<4> Response type and data exchange format.
+<5> Specifies that the `exchangeRate` operation is secured with the `acme-financial-oauth` security scheme.
+<6> Response type specification.
+<7> `acme-financial-oauth` security scheme specification.
+<8> Security scheme type. This information indicates that you must configure a {quarkus_guide_security_openid_connect_client_url}[Quarkus OpenId Connect Client (OIDC)] with name `acme_financial_oauth` to execute the operation.
+<9> Authentication flow to use, and flow information.
+--
+
+[NOTE]
+====
+You can find the full implementation of this service in the `serverless-workflow-oauth2-orchestration-quarkus/acme-financial-service` directory.
```

Comment:
This was done intentionally. 
At this particular point, I don't want the user to jump to another document and start reading the acme-financial-service.yml implementation. 
Instead, from the perspective of the SW, all the information you need about it is in the OpenApi specification. So the acme financial service is a kind of secondary actor. 
What I'm interested here is that the reader can see the spec, and continue to the "Configuration", so the note is added in this way, no clicks, no distraction, but an indicator where it is.

There might even be the case or readers that will never run, or clone the example.
Why?
Because the guide, even when not perfect, is structured in a way that you can get all the concepts by not having o see the repository with the example.

In fact, the only thing that is shown about acme is the yml file, the important thing. Readers that wants to go a step further, will clone the example, execute, etc.

Reply:
it is not important though, just my personal preference, when reading a guide, even if the link is mentioned before, would be nice to have a link at the point you're reading to quickly access it if needed.

feel free to ignore it :)